src/Symfony/Component/Validator/Constraints/JsonValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class JsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Json) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Json');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
```

Comment:
For consistency, a type check should be added

https://github.com/symfony/symfony/blob/00e5cd9a1c237e579e6327e9a66c512bf76f292a/src/Symfony/Component/Validator/Constraints/IpValidator.php#L39-L43